### ISSUE TITLE
Issue with merging split tiles

### ISSUE THREAD

**Issue by liuyadong** [ranghetti/sen2r#434]
**Issue description**

Hi, I don't know why but in some cases one Sentinel tile was split into two. Their non-NA parts are complimentary but they have the same extent. With sen2r both can be downloaded from SciHub but only one is available on Google Cloud. When merging with adjacent tiles, sen2r would use one of them ignoring the other even if both are downloaded. Any way to deal with this case?

**Reproducible example**

```r
library(sen2r)
library(sf)

point <- st_point(c(106.7, 37.5)) |> 
  st_sfc(crs=st_crs(4326))

lists1 <- s2_list(
  point,
  server='gcloud',
  time_interval=c("2021-01-06",'2021-01-07')
)

lists2 <- s2_list(
  point,
  server='scihub',
  time_interval=c("2021-01-06",'2021-01-07')
)

lists1

A named vector with 1 SAFE archives.
                                                    S2A_MSIL2A_20210106T034131_N0214_R061_T48SXG_20210106T055417.SAFE 
"gs://gcp-public-data-sentinel-2/L2/tiles/48/S/XG/S2A_MSIL2A_20210106T034131_N0214_R061_T48SXG_20210106T055417.SAFE/" 
The following attributes are included: mission, level, id_tile, id_orbit, sensing_datetime, ingestion_datetime, clouds, footprint, uuid, online.

lists2

A named vector with 2 SAFE archives.
                                     S2A_MSIL2A_20210106T034131_N0214_R061_T48SXG_20210106T070711.SAFE 
"https://apihub.copernicus.eu/apihub/odata/v1/Products('6d8a00c5-88ac-45fa-99a4-e09e636c7f73')/$value" 
                                     S2A_MSIL2A_20210106T034131_N0214_R061_T48SXG_20210106T055417.SAFE 
"https://apihub.copernicus.eu/apihub/odata/v1/Products('f7059ebf-c8b9-44af-aae4-6a0f858124e9')/$value" 
The following attributes are included: mission, level, id_tile, id_orbit, sensing_datetime, ingestion_datetime, clouds, footprint, uuid, online.
```

Here are what two tiles look like:

S2A_MSIL2A_20210106T034131_N0214_R061_T48SXG_20210106T055417.SAFE

![S2A2A_20210106_061_48SXGb_TCI_10](https://user-images.githubusercontent.com/7848019/154496222-5da8781f-d11e-4f95-91e2-437a02a9c7b9.jpg)

S2A_MSIL2A_20210106T034131_N0214_R061_T48SXG_20210106T070711.SAFE

![S2A2A_20210106_061_48SXGa_TCI_10](https://user-images.githubusercontent.com/7848019/154496270-ee00a17a-4981-41bb-ac45-7d3ab088cb4d.jpg)

**Comment by ranghetti** [comment]
Actually sen2r is able to deal with this case, e.g.
```r
sen2r(
  gui=FALSE,
  list_rgb="RGB432B",
  online = FALSE,
  timewindow = c("2021-01-06",'2021-01-07'), 
  extent_name = "issue434", 
  s2tiles_selected = "48SXG",
  clip_on_extent = FALSE,
  path_out = issue_dir, path_l2a = safe_dir, path_l1c = safe_dir
)

```

![image](https://user-images.githubusercontent.com/6874239/154521676-afc0a637-d107-40d3-b636-49f0b0156f95.png)

**Comment by liuyadong** [comment]
Aha thanks! It works well, but not for `list_prods = 'TCI'`.

**Comment by liuyadong** [comment]
There is still a missing part in this case:

```
sen2r(
  gui = FALSE,
  online = TRUE,
  server = 'scihub',
  order_lta = TRUE,
  timewindow = c("2021-02-15", "2021-02-16"),
  extent = "dingbian.gpkg",
  extent_name = 'dingbian',
  list_rgb = 'RGB432B',
  clip_on_extent = TRUE,
  extent_as_mask = TRUE,
  path_l1c = dir_safe,
  path_l2a = dir_safe,
  path_out = dir_issue
)
```
![S2A2A_20210215_061_dingbian_RGB432B_10](https://user-images.githubusercontent.com/7848019/154610455-1ca3ada9-ced4-4f8b-af80-52bd0d8a3636.jpg)

dingbian.gpkd is attached
[dingbian.zip](https://github.com/ranghetti/sen2r/files/8093855/dingbian.zip)

**Comment by liuyadong** [comment]
After deepening this issue, I found it is related to the scheme of sen2r adding suffix to split tiles, i.e. adding abc inversely according to the creation time. I don't know the reason of this design, but for gdalbuildvrt the order of tiles matters. For my specific case, adding suffix the other way perfectly solved this issue (no missing area). I'm wondering whether this solution applies to other cases as well, if so I kindly ask to upgrade the package. Thank you.

**Comment by ranghetti** [comment]
@liuyadong currently I have not the possibility to spent time on this package. Since you have deepened the problem, can you please detail in which way the code should be changed (I have not understood what you mean with "adding suffix the other way").  A code chunk would be the better way to understand. Thank you

**Comment by liuyadong** [comment]
By that I mean adding suffix in an ascending order, not inversely, according to creation_datetime. Doing so the latest tile would be on the top when merging, which might be a universal rule I guess. Here is the code for my case:

```r
library(sen2r)
library(fs)

dir_safe <- '.'
dir_ls(dir_safe, type='directory', glob='*20210215*')
#> S2A_MSIL2A_20210215T033831_N0214_R061_T48SXG_20210215T054829.SAFE
#> S2A_MSIL2A_20210215T033831_N0214_R061_T48SXG_20210215T073644.SAFE
#> S2A_MSIL2A_20210215T033831_N0214_R061_T48SYF_20210215T054829.SAFE
#> S2A_MSIL2A_20210215T033831_N0214_R061_T48SYF_20210215T073644.SAFE
#> S2A_MSIL2A_20210215T033831_N0214_R061_T48SYG_20210215T054829.SAFE
#> S2A_MSIL2A_20210215T033831_N0214_R061_T48SYG_20210215T073644.SAFE

sen2r(
  gui = FALSE,
  online = FALSE,
  s2tiles_selected = c('48SXG','48SYF','48SYG'),
  timewindow = c('2021-02-15', '2021-02-16'),
  clip_on_extent = FALSE,
  list_prods = 'BOA',
  outformat = 'VRT',
  path_l1c = dir_safe,
  path_l2a = dir_safe,
  path_out = 'out',
  path_tiles = 'tiles',
  thumbnails = FALSE
)

out <- 'out/BOA/S2A2A_20210215_061__BOA_10.vrt'
out |> 
  s2_thumbnails()
```

![S2A2A_20210215_061__BOA_10](https://user-images.githubusercontent.com/7848019/155276971-d924385c-9e7d-4804-822e-44f2ca2d2546.jpg)

```
(tiles <- dir_ls('tiles/BOA'))
#> tiles/BOA/S2A2A_20210215_061_48SXGa_BOA_10.vrt
#> tiles/BOA/S2A2A_20210215_061_48SXGb_BOA_10.vrt
#> tiles/BOA/S2A2A_20210215_061_48SYFa_BOA_10.vrt
#> tiles/BOA/S2A2A_20210215_061_48SYFb_BOA_10.vrt
#> tiles/BOA/S2A2A_20210215_061_48SYGa_BOA_10.vrt
#> tiles/BOA/S2A2A_20210215_061_48SYGb_BOA_10.vrt

tiles2 <- c(
  'tiles/BOA/S2A2A_20210215_061_48SXGb_BOA_10.vrt',
  'tiles/BOA/S2A2A_20210215_061_48SXGa_BOA_10.vrt',
  'tiles/BOA/S2A2A_20210215_061_48SYFb_BOA_10.vrt',
  'tiles/BOA/S2A2A_20210215_061_48SYFa_BOA_10.vrt',
  'tiles/BOA/S2A2A_20210215_061_48SYGb_BOA_10.vrt',
  'tiles/BOA/S2A2A_20210215_061_48SYGa_BOA_10.vrt'
  )
tiles2 |> 
  s2_merge() |> 
  s2_thumbnails()
```
![S2A2A_20210215_061__BOA_10](https://user-images.githubusercontent.com/7848019/155277218-cea671bf-26eb-407e-9a62-b60629645bfc.jpg)

**Comment by liuyadong** [comment]
@ranghetti I'm afraid the actual issue is related to the spherical geometry.

``` r
library(tidyverse)
library(sen2r)
#> Registered S3 method overwritten by 'geojsonsf':
#>   method        from   
#>   print.geojson geojson
#> Welcome to sen2r. To use the package from a GUI, launch
#>  > sen2r()
#> Documentation: https://sen2r.ranghetti.info
library(sf)
#> Linking to GEOS 3.10.2, GDAL 3.4.1, PROJ 8.2.1; sf_use_s2() is TRUE


safe <- 'S2A_MSIL2A_20210215T033831_N0214_R061_T48SYG_20210215T054829.SAFE'
meta <- safe |> 
  safe_getMetadata()

sf_use_s2(FALSE)
#> Spherical geometry (s2) switched off
meta |> 
  st_as_sf(wkt = 'footprint', crs=4326) |> 
  mutate(area = st_area(footprint)) |> 
  mutate(do_step3 = area < 109e3^2*units::as_units("m^2")) |> 
  st_drop_geometry() |> 
  select(area, do_step3)
#>                 area do_step3
#> 1: 11884897714 [m^2]    FALSE

sf_use_s2(TRUE)
#> Spherical geometry (s2) switched on
meta |> 
  st_as_sf(wkt = 'footprint', crs=4326) |> 
  mutate(area = st_area(footprint)) |> 
  mutate(do_step3 = area < 109e3^2*units::as_units("m^2")) |> 
  st_drop_geometry() |> 
  select(area, do_step3)
#>                 area do_step3
#> 1: 11879276691 [m^2]     TRUE
```

<sup>Created on 2022-02-23 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by ranghetti** [comment]
I reversed the order of merged tiles in b27e0f1b740dcfd5cd517726d6e87b2cb67f2c80, for now in development branch, you can use it by reinstalling the package with `remotes::install_github("ranghetti/sen2r", ref = "devel")`.
Spherical geometry is switched off in sen2r processing because of errors which occurred leaving it on; I will not modify this behaviour in order not to introduce further errors.

**Comment by liuyadong** [comment]
It works very well. Thank you Dr. Ranghetti!
